### DISCUSSION TITLE
Can't install ruby 3.1.3 on Ubuntu WSL

### DISCUSSION THREAD

**Discussion by NicolasMDuarte** [rbenv/ruby-build#2114] [Build failures] [answered]
It gives me the error below while installing version 3.1.3:

```
current directory: /home/duarte_nicolas/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rbs-2.7.0/ext/rbs_extension
make DESTDIR\\= sitearchdir\\=./.gem.20221216-14046-z50xuq sitelibdir\\=./.gem.20221216-14046-z50xuq install
make[1]: Entering directory '/home/duarte_nicolas/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rbs-2.7.0/ext/rbs_extension'
/home/linuxbrew/.linuxbrew/bin/gmkdir: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
make[1]: *** [Makefile:204: .sitearchdir.time] Error 127
make[1]: Leaving directory '/home/duarte_nicolas/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rbs-2.7.0/ext/rbs_extension'

make install failed, exit code 2
```

**Comment by mislav** [comment]
It looks like you have something called `gmkdir` installed with Homebrew and it might be causing trouble.

You can find out which formula provides gmkdir with: `ls -l /home/linuxbrew/.linuxbrew/bin/gmkdir`. That will show the symlink destination to a specific formula. Then, try `brew unlink <formula>`

**Comment by JuPlutonic** [comment, marked as answer]
Yep `ls -l`/`readlink`/`file` (or  easier `brew whence gmkdir` will help you).
It gives `coreutils`, so `brew unlink coreutils` will help with that building rbs c-extension problem.
For me it does help.

---
But I can't find the real cause of the error. Now, after checking what my PATH look like "/home/linuxbrew/.linuxbrew/sbin:/home/linuxbrew/.linuxbrew/bin:REST", I tring to reinstall all connected formulae on HOMEBREW like `binutils`, `gcc`, `ruby-build`, `glibc`. I have big issue with pg c-extension ~ libssl-dev(ubuntu 18.04) when bundler works so I think what all errors have the one source.

**Reply by NicolasMDuarte** [reply]
Hey, it´s gcc version 11.3.0 (Ubuntu 11.3.0-1ubuntu1~22.04) 😄

**Reply by NicolasMDuarte** [reply]
And thank you very much!! You and @mislav helped me a lot. I've been stuck here for a hole day and you both just saved me. ❤️

**Reply by JuPlutonic** [reply]
```bash
# This should work (rbs – ok) w/o coreutils unlinking:
brew reinstall coreutils # If it had unlinked
brew install gcc@12 #           gcc@8 – gcc@12 ... you need to choose one
CC=gcc-12 rbenv install 3.2.0 # gcc-8 – gcc-12 ... you need to choose one
```
The answer helps me on Ununtu 18.04 + rbenv install 3.2.0 + homebrew (gcc@12, RUBY_CONFIGURE_OPTS with openssl@3) \
But it doesn't helped me with rbenv install 3.1.3 (and with rbenv install 3.0.5) at all \
rbenv/asdf  based) s(t)ucked here.
> PS for rbenv I don't know how to get /rbs-2.7.0/gem_make.out **when I do rbenv install -kv 3.1.3**

>PPS Its impossible to use 3.0.5 IN UBUNTU + HOMEBREW environment (ruby is installed from sources to rbenv directory like this `./configure --prefix=/home/alessio/.rbenv/versions/3.0.5 --with-openssl-dir="/home/linuxbrew/.linuxbrew/opt/openssl@1.1"`), when you gem install _something_  you get \
`ERROR:  While executing gem ... (Gem::Exception)`\
`    OpenSSL is not available. Install OpenSSL and rebuild Ruby (preferred) or use non-HTTPS sources`
I'm thinking moving away not from Linux+Homebrew but from rbenv/ruby-build because `rvm pkg install openssl` may help to get working dev environment (3.0.4 produced by rvm wonderfully working in rbenv).

**Reply by alexeyr** [reply]
> But it doesn't helped me with rbenv install 3.1.3 (and with rbenv install 3.0.5) at all
rbenv/asdf based) s(t)ucked here.

I previously had similar problems, but this time `CC=gcc-13 CXX=g++-13 rbenv install 3.1.3` worked (`CXX` is needed for some gems I use and passing it only to `bundle install` didn't work).
